Classes/extra_functions.py

```diff
@@ -93,4 +93,14 @@ def get_rank_id(settings, name_id):
 
 
 def has_role(role_list, role_id):
-    return len([x for x in role_list if x.id == role_id]) > 0
\ No newline at end of file
+    return len([x for x in role_list if x.id == role_id]) > 0
+
+
+# Process the role_ladder into a usable list when called
+def role_id_index(self, bad_role=0):
```

Comment:
Self should not be passed into a function outside a class, self refers to the instance of the class that you are operating on. Like if you pass self into a function on the Officer class, self references that officer so that you can do self.id or self.is_on_duty.

You can check out the tutorial on classes in python for more information on self.

Reply:
Fixed in commit